plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableDataStoreTests.php

```diff
@@ -3198,4 +3198,45 @@ public function test_order_updated_webhook_delivered_once() {
 		remove_all_actions( 'woocommerce_webhook_process_delivery' );
 		remove_all_actions( 'woocommerce_webhook_should_deliver' );
 	}
+
+	/**
+	 * @testDox Check that functions in the datastore correctly hold and release coupons from the order.
+	 */
+	public function test_datastore_coupon_methods() {
+		$this->toggle_cot_authoritative( true );
+
+		$coupon = new \WC_Coupon();
+		$coupon->set_code( '10off' );
+		$coupon->set_discount_type( 'percent' );
+		$coupon->set_amount( 10.0 );
+		$coupon->set_usage_limit_per_user( 2 );
+		$coupon->save();
+
+		$product = WC_Helper_Product::create_simple_product( true );
+
+		WC()->cart->add_to_cart( $product->get_id(), 1 );
+		WC()->cart->add_discount( $coupon->get_code() );
+
+		$this->assertEquals( 0, $coupon->get_data_store()->get_usage_by_email( $coupon, 'user@woo.test' ) );
+
+		$order_id = WC()->checkout->create_order(
+			array(
+				'billing_email'  => 'user@woo.test',
+				'payment_method' => 'dummy'
```

Comment:
The linter [really, really wants](https://github.com/woocommerce/woocommerce/actions/runs/6903967483/job/18783779604?pr=41538) this to end in a comma:

```suggestion
				'payment_method' => 'dummy',
```